org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -1442,6 +1516,119 @@ private boolean isValid(@Nullable WorkspaceFolder wsFolder) {
 
 	}
 
+	/**
+	 * Resource listener that translates Eclipse resource change events into LSP
+	 * file watch events and dispatches them if the language server is still active
+	 */
+	private final class WatchedFilesListener implements IResourceChangeListener {
```

Comment:
move to its own class file in the internal package?

Reply:
For that LanguageServerWrapper is in the wrong package. I would not want to pollute the root package with a "random" WatchedFilesListener class file that is an internal listener implementation (like LSFileBufferListener or WorkspaceFolderListener) of LanguageServerWrapper. If LoC of LanguageServerWrapper is a concern we should better think about moving all the generic utility methods like getRootCauseMessage/getThrowableMessage/disconnectTextFileBuffer/supportsWorkspaceFolders etc. out of LanguageServerWrapper but I would do this in a dedicated PR focused on refactoring/cleanup of the class.